includes/admin/debug-tools/class-wcs-notifications-debug-tool-processor.php

```diff
@@ -0,0 +1,311 @@
+			// phpcs:disable WordPress.DB.PreparedSQLPlaceholders.ReplacementsWrongNumber
+				$wpdb->prepare(
+					"SELECT 
+								id
+							FROM {$wpdb->prefix}wc_orders 
+							WHERE type='shop_subscription'
+							AND status IN ($placeholders)
+							ORDER BY id ASC
+							LIMIT %d
+							OFFSET %d",
+					...$args
+				)
+			);
+		} else {
+			$subscriptions_to_process = $wpdb->get_col(
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			// phpcs:disable WordPress.DB.PreparedSQLPlaceholders.ReplacementsWrongNumber
+				$wpdb->prepare(
+					"SELECT 
+								ID
+							FROM {$wpdb->prefix}posts 
+							WHERE post_type='shop_subscription'
+							AND post_status IN ($placeholders)
+							ORDER BY ID ASC
+							LIMIT %d
+							OFFSET %d",
+					...$args
+				)
+			);
+		}
+
+		// Reset the tool state if there are no more subscriptions to process.
+		if ( empty( $subscriptions_to_process ) ) {
+			$this->delete_tool_state();
+		}
+
+		return $subscriptions_to_process;
+	}
+
+	/**
+	 * Process data for the supplied batch.
+	 *
+	 * This method should be prepared to receive items that don't actually need processing
+	 * (because they have been processed before) and ignore them, but if at least
+	 * one of the batch items that actually need processing can't be processed, an exception should be thrown.
+	 *
+	 * Once an item has been processed it shouldn't be counted in 'get_total_pending_count'
+	 * nor included in 'get_next_batch_to_process' anymore (unless something happens that causes it
+	 * to actually require further processing).
+	 *
+	 * @throw \Exception Something went wrong while processing the batch.
+	 *
+	 * @param array $batch Batch to process, as returned by 'get_next_batch_to_process'.
+	 */
+	public function process_batch( array $batch ): void {
+		// This is a bit unnecessary. Perhaps convert `update_status` to static to avoid instantiating the class?
+		$subscriptions_notifications = new WCS_Action_Scheduler_Customer_Notifications();
+
+		foreach ( $batch as $subscription_id ) {
+			$subscription = wcs_get_subscription( $subscription_id );
+			$subscriptions_notifications->update_status( $subscription, $subscription->get_status(), null );
+
+			// Update the subscription's update time to mark it as updated.
+			$subscription->set_date_modified( time() );
+			$subscription->save();
+		}
+
+		// Update tool state.
+		$state                = $this->get_tool_state();
+		$state['last_offset'] = isset( $state['last_offset'] ) ? absint( $state['last_offset'] ) + count( $batch ) : count( $batch );
+		$this->update_tool_state( $state );
+	}
+
+	/**
+	 * Default (preferred) batch size to pass to 'get_next_batch_to_process'.
+	 * The controller will pass this size unless it's externally configured
+	 * to use a different size.
+	 *
+	 * @return int Default batch size.
+	 */
+	public function get_default_batch_size(): int {
+		return 20;
+	}
+
+	/**
+	 * Start the background process for coupon data conversion.
```

Comment:
```suggestion
	 * Start the background process for batch processing subscription notifications updates.
```

Looks like a copy-paste.